lib/msf/core/db_manager/payload.rb

```diff
@@ -9,18 +9,22 @@ def create_payload(opts)
       end
 
       wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework)
-      wspace.payloads.create!(opts)
+      Mdm::Payload.create!(opts)
```

Comment:
I don't think this is going to play well with the remote data service.

Reply:
This looks like it works OK. Tested with remote data service and local pg connection.